### DISCUSSION TITLE
Unstable / long compile time and how to improve

### DISCUSSION THREAD

**Discussion by ylazy** [BowlerHatLLC/vscode-as3mxml#759] [General]
The same code, sometimes it's fast (~5s) but sometimes it's very slow (~50s) to compile on my laptop (Asus Vivo Book).
After so many times trying to determine where the problem is, I finally did some changes to get it more stable.

- [Turn off indexing](https://answers.microsoft.com/en-us/windows/forum/all/how-do-you-turn-off-indexing-in-windows-10/8e8829d3-cad1-46d7-a523-2cc9d6ed646a)
- [Disable System Acceleration with Intel® Optane™ Memory](https://www.intel.com/content/www/us/en/support/articles/000024626/memory-and-storage/intel-optane-memory.html)
- [Turn off System Restore](https://www.thewindowsclub.com/how-to-turn-off-system-restore-in-windows)
- Increase your physical memory (RAM) if needed, and increase the maximum heap size: `"as3mxml.languageServer.jvmargs": "-Xmx2g",`
- Restart your computer.

My app then take only ~5 - 6s to compile, and much more stable.

Hope this helps someone.

Thanks!

**Comment by Adolio** [comment]
I don't know where could come from this big variation, maybe a cache that is cleaned up time to time?

However on my side I sped up compile time by avoiding embedding assets (in debug builds at least) - this was taking most of my compilation time.
